src/main/java/nl/tudelft/contextproject/Main.java

```diff
@@ -91,10 +121,13 @@ public static void setInstance(Main main) {
 	public boolean setController(Controller c) {
 		if (c != controller) {
 			if (controller != null) {
-				stateManager.detach(controller);
+				getStateManager().detach(controller);
 			}
 			controller = c;
-			stateManager.attach(controller);
+			
+			if (controller != null) {

```

Comment:
No, @ChielBruin is correct.
The first line checks the old controller.
Then `controller = c` makes it so that `controller` points to the new controller.
Then the second check is for the `controller`, which is the new controller.